### PR TITLE
Add tests for sauron plugin

### PR DESCRIPTION
**Pull request by sip-21**
* Added tests for Sauron plugin using:
  - Esplora API
  - MutinyNet API (mempool.space)
  - Tor proxy configuration

### CODE REVIEW

**Comment by sip-21** [comment]
Tests are probably not yet complete, but provide a base.

File: sauron/sauron.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 plugin = Plugin(dynamic=False)
 plugin.sauron_socks_proxies = None
-plugin.sauron_network = "test"
+plugin.sauron_network = "bitcoin"
```
Comment (chrisguida) [inline sauron/sauron.py]: Why this change?

Seems like not a great idea to make mainnet the default?

**Comment by ca-ruz** [comment]
@sip-21 

Great! All tests for Blockstream Esplora passed. I also ran them locally and they also passed. Would be great if you can write some tests for Mutinynet.

File: sauron/sauron.py (hunk 1 above)
Comment (sip-21) [reply sauron/sauron.py]: Upss, this was unintentional. A leftover from testing.
Fixed.

**Comment by sip-21** [comment]
I ran `ruff` on the code and did some more cleanups.

**Comment by chrisguida** [comment]
@ca-ruz let's merge this and figure out the tests on the upstream PR

**Comment by ca-ruz** [comment]
Awesome @sip-21 Thanks for writing the tests for Sauron! Very thorough. Merged.
